### PR TITLE
8296158: Refactor the verification of CDS region checksum

### PR DESCRIPTION
**Pull request by iklam**
Consolidate the two old functions

- `FileMapInfo::region_crc_check(char* buf, size_t size, int expected_crc)`
- `FileMapInfo::verify_region_checksum(int i)`

Into a single function `FileMapRegion::check_region_crc()`.

Also performed some minor renaming and removed a useless API.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296158](https://bugs.openjdk.org/browse/JDK-8296158): Refactor the verification of CDS region checksum

### Reviewers
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to [3ad8d838](https://git.openjdk.org/jdk/pull/12424/files/3ad8d838844fd66bca2e2ad69667f6090b1f2183)
 * [Matias Saavedra Silva](https://openjdk.org/census#matsaave) (@matias9927 - Author) ⚠️ Review applies to [10dcd0b1](https://git.openjdk.org/jdk/pull/12424/files/10dcd0b17507ebc5b33fb5a5ac4278aa4e9a548e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12424/head:pull/12424` \
`$ git checkout pull/12424`

Update a local copy of the PR: \
`$ git checkout pull/12424` \
`$ git pull https://git.openjdk.org/jdk pull/12424/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12424`

View PR using the GUI difftool: \
`$ git pr show -t 12424`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12424.diff">https://git.openjdk.org/jdk/pull/12424.diff</a>

</details>

### CODE REVIEW

**Review by calvinccheung — APPROVED** [review]
Code refactoring looks good.

**Review by matias9927 — CHANGES_REQUESTED** [review]

File: src/hotspot/share/cds/filemap.cpp [hunk 1]
```diff
@@ -1592,6 +1592,27 @@ BitMapView FileMapRegion::ptrmap_view() {
   return bitmap_view(false);
 }
 
+bool FileMapRegion::check_region_crc() const {
+  // This function should be called after the region has been properly
+  // loaded into memory via FileMapInfo::map_region() or FileMapInfo::read_region().
+  // I.e., this->mapped_base() must be valid.
+  if (!VerifySharedSpaces) {
```
[the remark this answers is not in the document]
Comment (iklam) [reply src/hotspot/share/cds/filemap.cpp]: Thanks for the comment. I fixed the code as you suggested.

**Review by matias9927 — APPROVED** [review]
LGTM!

**Comment by iklam** [comment]
Thanks @matias9927 and @calvinccheung for the review
/integrate

**Pull request closed without merging**